### PR TITLE
New Script - MultiString Search and Replace in Measures

### PR DESCRIPTION
**Pull request by danmeissner**
Forget the Add files via upload and Delete first actions, I attached the wrong file and then deleted it. 

The last commit of "New Script;" has the new script that does paired strings for "search for this and replace with that" for multiple in an array on each selected measure expression.

### CODE REVIEW

File: Intermediate/MultiReplaceInMeasures.csx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Title: Multi-string Search and Replace in any selected measure
+ * 
+ * Author: Dan Meissner
+ * 
+ * This script will loop through a matched pair of strings held in a 2-D array 
+ * and search and replace a FromString and ToString pair in the currently selected measures.
```
Comment (JamesDBartlett3) [inline Intermediate/MultiReplaceInMeasures.csx:7]: How would you feel about updating this section to read like this?
> This script iterates through all currently-selected measures, using the matched pairs of strings 
> in the 2-D array named "ReplacementPair" below as a template, replacing all instances of "FromStringX" 
> in the current measure's DAX expression with the corresponding "ToStringX"

File: Intermediate/MultiReplaceInMeasures.csx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// Replace Text Strings Pairs that appear in selected measures
+
+    // Update the value of the text arrays for your desired usecase. Make sure to list structure the list as:
+    // var ReplacementPair  = new string[,] {{"FromString1","ToString1"},{"FromString2","ToString2"},{"FromString3","ToString3"}};
+	
+	// Add as many From and To pairs to the array as needed. 
+	// (technically C# has a 2GB memory size and 4 Billion array item limit, but... really...)
+
+    // If the string you are either searching for or replacing with contains a double quote " then you need to 'escape it' by 
+    // proceeding it with a backslash \ (as in \") to have that quote character wihtin the respective text string
```
Comment (JamesDBartlett3) [inline Intermediate/MultiReplaceInMeasures.csx:20]: Looks like there are a few small typos here. 

> ~~proceeding~~ preceding it with a backslash ~~\ (as in \\")~~ (as in `\"`) to have that quote character ~~wihtin~~ within the respective text string

File: Intermediate/MultiReplaceInMeasures.csx (hunk 1 above)
Comment (danmeissner) [reply Intermediate/MultiReplaceInMeasures.csx:7]: Much better said :)  Looking through some of the comments I put in there months ago, it must have been late at night.  Thanks for looking at this with a second, objective set of eyes.

File: Intermediate/MultiReplaceInMeasures.csx (hunk 2 above)
Comment (danmeissner) [reply Intermediate/MultiReplaceInMeasures.csx:20]: Again, thanks for catching these.

**Comment by JamesDBartlett3** [comment]
@danmeissner the file appears unchanged from my end. Did you make a new commit?

**Comment by JamesDBartlett3** [comment]
@danmeissner it looks like @MikeCarlo just merged the PR as-is.

**Comment by JamesDBartlett3** [comment]
@danmeissner I just updated the file myself. It's now published in the main branch [here](https://github.com/PowerBI-tips/TabularEditor-Scripts/blob/main/Intermediate/Search%20and%20Replace%20Matched%20Pairs%20of%20Strings%20in%20Selected%20Measures.csx).

**Pull request merged**